### PR TITLE
Items That Make Players Into Antags

### PR DESCRIPTION
**Pull request by Shadowmech88**
I've added a couple new items that can turn players into antags with some semblance of being IC, rather than literally being given the power by the gods, in the same vein as the Syndicate Uplink wish granter we currently have.

The changeling vial, which actually has nothing in common with actual vials except its appearance. When used, the user crushes it in their hand, and the contents attack them and inhabit their body. The vial is one-use only.

The stone mask, which turns a person into a vampire when they put it on and splash blood on it. It is one-use by default, but there is an infinite use version that can be spawned.

Neither item can be used by someone that is already that antag type. Both items can only be used by humans (this includes our various player races.)

New vampires created via the mask receive 300 blood to start with, and new changelings created via the vial receive 10 genomes.
This is done because any player that uses these will most likely not have found them at round start, and so would be at a bit of a disadvantage otherwise due to getting such a late start.

### CODE REVIEW

**Comment by MusgardTiger** [comment]
People will race for these things no matter where you put it, I don't think it is a really good idea to put antag making items inside the game.

**Comment by Shadowmech88** [comment]
The Syndicate Uplink already exists as part of a space vault.
In any case this PR doesn't add them to the map. An admin could even spawn them as part of adminbus if he wished.

**Comment by TheFriggenSawbones** [comment]
its a cool idea but wouldn't it end up being used for a bus or two and forgotten afterwards

**Comment by MusgardTiger** [comment]
I'll be okay with it if you want to put it in vaults doe. I didn't word my previous post very well. Don't add it to xenoarch or asteroid doe.

**Comment by Shadowmech88** [comment]
If other people want to make vaults with these in them, they're welcome to. I just wanted to get the code in so that other people have that opportunity.

**Comment by kol1th** [comment]
i like it

**Comment by BestName** [comment]
> New vampires created via the mask receive 300 blood to start with, and new changelings created via the vial receive 10 genomes.

But why? If you're going out of your way to turn yourself into an antag then shit should be tough.
Fuck this dumb handholding.

**Comment by carshalash** [comment]
10 genomes is a bit much, the blood seems fine though.

Blob item when

**Comment by jknpj** [comment]
These kind of items should have a downside, like the cursed dice. Maybe a chance of killing the user?

**Comment by Intigracy** [comment]
Yeah 10 genomes is an insane amount.

:+1: for subtle jojo reference though.

**Comment by Shadowmech88** [comment]
Is it though? Roundstart changelings start with 5 genomes, don't they?

**Comment by PJB3005** [comment]
> Tell people to NOT FUCKING ADD TO THE CANCER THAT IS GAMEMODES.
> They don't fucking listen.

**Comment by Intigracy** [comment]
So just slap DNM on it until your refactor is done

**Comment by Shadowmech88** [comment]
How's that refactor coming along, by the way?

**Comment by ghost** [comment]
Wouldn't the mask come with a chance of it killing the user with the spikes?

There needs to be some sorta downside to trying to activate these, like the goo in the vial just dying or 'falling to the floor and dissipating', or critical failure it latches to your face, body rejects it, you explode.

**Comment by Shadowmech88** [comment]
I used the Synducate Uplink as a template for the functionality of these. It has no downside or risk for turning people into traitors.

**Comment by carshalash** [comment]
Oh yeah. Death probability was something i needed.

**Comment by Blithering** [comment]
@Shadowmech88 If by the syndicate uplink you mean the beacon you can find through archaeology, that thing has a chance to blow up on use, killing the user and anyone nearby. I would consider that a downside.

**Comment by Shadowmech88** [comment]
I was talking specifically about the one in the Syndicate Outpost vault. I've never had it blow up on me.

**Comment by aceedex** [comment]
The one on the syndie vault does blow up from personal experience, but I have the suspicion that it has something to do with your character's NT relations

**Comment by Blithering** [comment]
Maybe have the stone mask have a chance to cause a lot of bleed and brute damage to the head region when it fails, and have the vial have a chance to dissolve the person's hand when crushed.

**Comment by gbasood** [comment]
I think the bottom line with these items is that they should be bus ONLY and if we start seeing these popping up via vaults or whatever we might have to remove them later.

As they are now though they'll be okay

**Comment by Shadowmech88** [comment]
Personally, I don't think having them in vaults would be an issue considering we've already got a vault that turns people into traitors.
But anyway, that's outside the scope of this PR. All this PR does is make them available for bus.

**Comment by gbasood** [comment]
@Shadowmech88 

> already have a vault that turns people into traitors

What is it and who the fuck though it was a good idea

**Comment by Shadowmech88** [comment]
@gbasood 
#8873.
The syndicate uplink in the southwest corner can make you into a traitor, I've personally used it.

**Comment by Zth--** [comment]
Is there a list of admin-only spawnable items? If not, most of stuff like this will be forgot in time

File: code/modules/clothing/masks/stone.dm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/obj/item/clothing/mask/stone/proc/spikes()
+	icon_state = "stone_spikes"
+	spikes_out = 1
+	if(istype(loc, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = loc
+		H.visible_message("<span class='warning'>Stone spikes shoot out from the sides of \the [src]!</span>")
+		if(H.wear_mask == src) //the mob is wearing this mask
+			if(H.mind)
+				if(!(H.mind in ticker.mode.vampires)) //They are not already a vampire
```
Comment (PJB3005) [inline code/modules/clothing/masks/stone.dm]: There are procs like `isvampire()`

File: code/game/objects/items/changeling_vial.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	icon_state = "secure_vial"
+	flags = FPRINT
+	w_class = 1
+	unacidable = 1
+	var/genomes_to_give = 10 //seeing as the new changeling won't have had a whole round to prepare, they get some genomes free
+
+/obj/item/changeling_vial/attack_self(mob/user as mob)
+	if(istype(user, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = user
+		if(!(H.mind in ticker.mode.changelings))
```
Comment (PJB3005) [inline code/game/objects/items/changeling_vial.dm]: `ischangeling()`

File: code/game/objects/items/changeling_vial.dm [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+			user.visible_message("<span class='warning'>As [user] crushes the vial, a mass of black goo leaps at \his face!.</span>","<span class='warning'>As you try to remove the cap, you crush the vial in your hand!</span> <span class='danger'>A mass of black goo leaps at you from the vial!</span>")
+			playsound(get_turf(src), "shatter", 20, 1)
+			ticker.mode.changelings += H.mind
```
Comment (ComicIronic) [inline code/game/objects/items/changeling_vial.dm]: Check there isn't a proc for all of this , because it looks very copy/pasted and that's bad.

Comment (gbasood) [inline code/game/objects/items/changeling_vial.dm]: HE HAS RETURNED

Comment (Intigracy) [inline code/game/objects/items/changeling_vial.dm]: ~~There's not a proc for this yet, no.~~

Comment (ComicIronic) [inline code/game/objects/items/changeling_vial.dm]: Is this copied from antag code?
If so, make it a proc.

File: code/modules/clothing/masks/stone.dm [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+					to_chat(H, "<span class='danger'>The mask's stone spikes pierce your skull and enter your brain!</span>")
+					ticker.mode.vampires += H.mind
```
Comment (ComicIronic) [inline code/modules/clothing/masks/stone.dm]: Same yada yada for vampire-making proc here.

File: code/game/objects/items/changeling_vial.dm (hunk 3 above)
Comment (Shadowmech88) [inline code/game/objects/items/changeling_vial.dm]: I actually copied it from the admin antag panel where you can make someone into various antag types.

If I make it a proc, where should I put the proc? `/mob/living/carbon/human`?

Comment (PJB3005) [inline code/game/objects/items/changeling_vial.dm]: Game mode or global.

Comment (Shadowmech88) [inline code/game/objects/items/changeling_vial.dm]: Should I bother yet? This has no chance of getting merged until @Intigracy is done with antag datums, right?

Comment (Intigracy) [inline code/game/objects/items/changeling_vial.dm]: /mob/proc/make_changeling() exists

Though I would prefer you not.

**Comment by clusterfack** [comment]
Fix this up and we'll merge it we're not waiting on antag datums right now since the window passed.

Also I want to see that mask as a xenoarch large artifact or as a prize in a really spooky jojo shrine vault/away mission

**Comment by Shadowmech88** [comment]
It'll be a few days before I can fix it up, but I'll get there.

**Comment by DeityLink** [comment]
dunno about the vial but the stone mask is a long overdue item.

**Comment by jknpj** [comment]
If you want them to be part of xenoarch, please put them INSIDE a large artifact(ex: the commstone artifact). Masks and vials aren't big enough to be considered LARGE artifacts, also, muh immulsions.

**Comment by DeityLink** [comment]
Nah, he can just add them to the happiest mask artifact, like, give the artifact a chance to have either one appearing instead.

**Comment by ihadtoregisterforthis** [comment]
Currently anyone with a diamond drill is a xenoarch, so adding these as random drop is potentially making them really common.
Large artifacts only appear if you do it very carefully.

**Comment by jknpj** [comment]
Consider the following: Just yesterday i've dug 5 Happiest Masks in a single shift.

**Comment by DeityLink** [comment]
oh right, happiest masks are small artifacts. forgot about that.

that's retarded.

**Comment by clusterfack** [comment]
An item that makes you a vampire and gives you ton of blood should not be anything less than a large artifact in itself

**Comment by PJB3005** [comment]
Consider the following: a couple of days ago I dug up 4 large artifacts all on my own.

**Comment by ihadtoregisterforthis** [comment]
Which might have been all the large artifacts dug that week

**Comment by Shadowmech88** [comment]
All the code problems have been fixed as well as the conflicts.

Once again ready for merge.

**Comment by jknpj** [comment]
Was the stone mask added as a large artifact too?

**Comment by Shadowmech88** [comment]
Not in this PR, no.

**Comment by blobbernaut** [comment]
Blob item when

**Pull request merged**